### ISSUE TITLE
Make .tex source of paper available

### ISSUE THREAD

**Issue by dfm** [openjournals/joss#132]
As mentioned in openjournals/joss-reviews#24, it would be great to make the .tex source of the compiled paper available (maybe it is and I'm just blind...) to enable things like uploading the paper to arXiv.

Thanks for making all this work!

**Comment by arfon** [comment]
We don't actually persist this anywhere currently but we could. We generate the PDF using this command: https://github.com/openjournals/whedon/blob/master/lib/whedon/processor.rb#L152-L171, the Pandoc tex template is here https://github.com/openjournals/whedon/blob/master/resources/latex.template

**Comment by kyleniemeyer** [comment]
I might be missing something, but what's the reason for uploading a JOSS paper to arXiv? The PDF generated by JOSS is citable with a DOI, and should persist.

**Comment by danielskatz** [comment]
I was wondering about that too.  Is there a reason a very short arXiv paper with an arXiv URL is better than a JOSS paper that has a DOI and is indexed?

**Comment by dfm** [comment]
@arfon: Yeah. I ran the pandoc command with output `paper.tex` and that worked but it might be convenient to add that as part of the build process. Your call!

@kyleniemeyer: In astronomy, papers pretty much don't exist unless they're on arXiv so I wanted to post it there to make it available to the community (and advertise JOSS 😄).

**Comment by dfm** [comment]
Note: people would cite JOSS but they would find the paper through arXiv.

**Comment by kyleniemeyer** [comment]
Gotcha, sounds good.

**Comment by arfon** [comment]
@dfm - I'm just revisiting this. What would you need for an arXiv submission - would it simply be the `paper.md` file compiled into a `paper.tex` file? We could certainly add this as an output and commit it to the `joss-papers` repo: https://github.com/openjournals/joss-papers/tree/master/joss.00024

**Comment by ngoldbaum** [comment]
@arfon I'm looking at this as well for the paper I'm submitting.

We'd need `paper.tex` file along with a way to get the `.sty` file to get the styling of the generated arxiv submission to match the JOSS internal styling. That's not necessary of course, I could just use a regular article document style, but that makes the final paper on the arxiv look less nice and professional.

Here's the details about how to upload a LaTeX document to the arxiv: https://arxiv.org/help/submit_tex

**Comment by labarba** [comment]
… and also the .bbl file, if that is generated from the .bib — arXiv will process this and add the bibliography.

**Comment by arfon** [comment]
Max Ogden has a good write up here https://gist.github.com/maxogden/97190db73ac19fc6c1d9beee1a6e4fc8 of how to do this.

I think you need to modify the [Whedon pandoc command](https://github.com/openjournals/whedon/blob/master/lib/whedon/processor.rb#L110-L133) to generate LaTeX as an alternative output. I'm not sure if that will produce `.bbl` files but it's a start.

**Comment by mattpitkin** [comment]
This may not be the best place to ask, but: I was just wondering if anyone has successfully submitted a JOSS paper to arXiv? I tried submitting [my paper](http://joss.theoj.org/papers/10.21105/joss.00538) to the arXiv today, but just got this email from them:

> Dear arXiv user,
> 
> arXiv only accepts complete, self-contained, research article submissions in a format appropriate for publication in a conventional journal. 
> 
> Your submission did not appear to be complete and, as a result, has been removed. Please feel free to resubmit a complete paper.
> 
> For more information, see: 
> ref: http://arxiv.org/help/primer 
> ref: http://arxiv.org/help/moderation 
> 
> Regards, 
> arXiv admin

**Comment by ngoldbaum** [comment]
It looks like @dfm ran into similar issues back in 2016, since corner.py never got an arxiv submission as far as I can see:

http://adsabs.harvard.edu/abs/2016JOSS....1...24F

**Comment by danielskatz** [comment]
Someone would have to contact arxiv to be sure, but my guess is that the 1-2 page typical JOSS paper is not something that arxiv is likely to consider sufficiently weighty/lengthy.

**Comment by mattpitkin** [comment]
@ngoldbaum yeah, I'd noticed that for corner.py too.

@danielskatz I'd guess so. It's a shame that arXiv won't accept JOSS papers, but I suppose it's not too much of a big deal. As @dfm mentioned before, it would be nice if they would accept them to advertise and provide more exposure for JOSS, but it's probably not worth pushing too hard.

**Comment by ngoldbaum** [comment]
I'm planning to submit a more substantial paper (~10 pages typeset) this week or next week. I will comment here about whether it gets through the arxiv moderation.

**Comment by ngoldbaum** [comment]
I asked  Steinn Sigurdsson (the arxiv scientific director) [on twitter](https://twitter.com/steinly0/status/1004092466667118592) about this. He pointed me to [the moderation policy](https://arxiv.org/help/moderation) which notes:

>  Abstract-only submissions, presentations, book announcements, book reviews, submissions without references, calls for papers, or advertisements may be removed.

So I guess that's the issue here. I expect a longer paper would make it past the moderators.

**Comment by mattpitkin** [comment]
@mhucka - I've appealed my arXiv rejection with the following email:

> Dear arXiv-moderation,
> 
> My submission submit/XXXXXXX was rejected and I presume this was due to the paper being very short. This is a paper that has been peer reviewed and accepted by the Journal of Open Source Software (JOSS) http://joss.theoj.org/ - which, although not a conventional journal, is becoming quite widely used, and may in the future be partnering with the American Astronomical Society. Typically, papers for this journal are very short, and as such what is included in the arXiv abstract field may represent the bulk of the paper, but they are complete and reviewed.
> 
> Would it be possible to reconsider this rejection for this article, and look at allowing JOSS papers for submission in the future?
> 
> Regards,
> 
> Matt Pitkin

I'll let you know what response I get.

**Comment by ngoldbaum** [comment]
My paper got posted: https://arxiv.org/abs/1806.02417

I ended up using @mattpitkin's makefile, which was very useful (and could probably be integrated into the JOSS submission instructions).

**Comment by mattpitkin** [comment]
I'm happy to report that the moderators accepted my appeal and my JOSS paper is now on [arXiv](https://arxiv.org/abs/1806.07809) (as is [another](https://arxiv.org/abs/1806.07442) JOSS paper I'm author on :smile:).

**Comment by SuperKam91** [comment]
In case this is any use to anyone, I submitted the .pdf of the JOSS paper, and appealed to arXiv to let me use this for the submission rather than the LaTeX files, due to the standard JOSS procedure not giving access to those files. They accepted my appeal and so I uploaded the .pdf to arXiv

**Comment by jonasrauber** [comment]
Has anything changed or is the above Makefile still the recommended workaround?
I'd prefer to submit LaTeX to arXiv rather than the PDF, if possible.

**Comment by arfon** [comment]
Nothing has changed sorry.

**Comment by flokno** [comment]
Simply providing the `.tex` file would be great. I was successful with using the Makefile, but it is slightly outdated and this overall felt like an unneccesary exercise.

**Comment by flokno** [comment]
FYI: My preprint got rejected even after appealing with a reference to [arxiv's policy on short works](https://blogs.cornell.edu/arxiv/2019/06/21/policy-on-short-works/).

**Comment by s-goldman** [comment]
I was able to reproduce my JOSS paper using latex. It took some tinkering, but they accepted the latex version. [https://arxiv.org/abs/2010.03031v1](https://arxiv.org/abs/2010.03031v1)

**Comment by dfm** [comment]
For those who continue to stumble across this:

@rodluger and I worked out yet another way to generate the .tex file for a JOSS manuscript using GitHub actions. I'm sure it would be possible to use [the official action](https://github.com/openjournals/openjournals-draft-action), but the way that we're doing it is (assuming the manuscript is in a directory called `joss`):

```yaml
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: TeX
        uses: docker://openjournals/paperdraft:latest
        with:
          args: joss/paper.md --to=latex --output=paper.tex
        env:
          GIT_SHA: $GITHUB_SHA
          JOURNAL: joss
      - name: PDF
        uses: docker://openjournals/paperdraft:latest
        with:
          args: joss/paper.md
        env:
          GIT_SHA: $GITHUB_SHA
          JOURNAL: joss
      - uses: actions/upload-artifact@v2
        with:
          path: joss
```

See it in action here: https://github.com/rodluger/starry_process

**Comment by ashleychontos** [comment]
The above workflow created a .tex file for me, thanks a bunch @dfm and @rodluger! I was curious if we can somehow apply this workflow to include the [most up-to-date draft](https://github.com/openjournals/joss-papers/blob/joss.03331/joss.03331/10.21105.joss.03331.pdf) in the review process?

**Comment by arfon** [comment]
@tarleb – might be good to adapt this Action to optionally include a .tex output https://github.com/openjournals/openjournals-draft-action ?

> I was curious if we can somehow apply this workflow to the most up-to-date draft in the review process?

@ashleychontos – the content in the `joss-papers` repo [you link to above](https://github.com/openjournals/joss-papers/blob/joss.03331/joss.03331/10.21105.joss.03331.pdf) is mostly for operating the journal so I don't really want to add .tex outputs there. That said, I'm definitely interested in making it easier for authors to generate a .tex version of their paper for their own needs.

**Comment by ashleychontos** [comment]
@arfon oh whoops, yeah that definitely makes sense. Like many others in this thread, I was just having issues uploading the current draft to arXiv. I imagine adding that action would definitely be helpful for others in the future

**Comment by tarleb** [comment]
I've made a small change to the underlying Docker image; passing `-k` as the first argument now ensures that the `paper.tex` is placed next to the `paper.pdf`. So the following should work:

``` yaml
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: TeX
        uses: docker://openjournals/paperdraft:latest
        with:
          args: '-k joss/paper.md'
        env:
          GIT_SHA: $GITHUB_SHA
          JOURNAL: joss
```

**Comment by zonca** [comment]
thanks,

I just submitted a JOSS paper to Arxiv, here my steps:

* modified github action and saved all artifacts: https://github.com/galsci/pysm/commit/9c91011133329877df685e0f293b2f856a74eee8
* downloaded JOSS logo from https://github.com/openjournals/whedon/blob/master/resources/joss/logo.png
* edited the `tex` file to point to local `logo.png`
* uploaded `paper.tex` `paper.bib` `logo.png` to Overleaf
* downloaded `output.bbl` from Overleaf, renamed to `paper.bbl`
* uploaded `paper.tex`, `logo.png` and `paper.bbl` to arxiv

**Comment by ashleychontos** [comment]
I was just able to finally submit to the arXiv based on @zonca 's detailed instructions (^^). Thank you so much @zonca 🙌

**Comment by rkurchin** [comment]
One note on @zonca's very nice clear suggestion above: It's not quite so simple if the repo isn't hosted on GitHub (e.g. it's on GitLab, as in https://github.com/openjournals/joss-reviews/issues/4040) – to me, this is an argument to maybe just make editorialbot have an option to generate and output the .tex directly...

**Comment by szhorvat** [comment]
@arfon I would be great for JOSS to make a clear statement about arXiv submissions. The [preprint policy](https://joss.readthedocs.io/en/latest/submitting.html#preprint-policy) states that preprints are okay, but in practice there are many obstacles to submitting to arXiv, the most prominent preprint server.

There are _workarounds_ above—I think it's fair to call them workarounds rather than solutions—but no officially supported way to produce a format that is suitable for arXiv. While JOSS is a software journal, many authors are still primarily researchers and not software developers. Using these workaround will not be trivial for many people, and having to jump through this many technical hoops just to submit a preprint is quite unpleasant.

There are several possibilities:

 - If JOSS wants to support preprint submissions _with JOSS's official template_, there should be a supported way to retrieve the TeX file.
 - JOSS many want to discourage using the official template (and JOSS logo), and ask authors to convert to TeX on their own. If so this should be made clear.
 - JOSS may want to discourage arXiv submissions due to the not-so-positive reactions by some moderators, reported in a few comments above. I assume this is not the case though.

I feel that the current approach by JOSS is neither here nor there. Using the template is not technically prevented, but in practice it's much more difficult than it should be. It would be great if you could clarify the situation :-)

**Comment by szhorvat** [comment]
I should note that adding `-k` to the command suggested [here](https://github.com/openjournals/inara#usage-via-docker) does not work, and it's not clear to me how to make use of @tarleb's [improvement](https://github.com/openjournals/joss/issues/132#issuecomment-855741414).

**Comment by arfon** [comment]
Thanks for the summary statement here @szhorvat. While I've not canvassed opinion widely amongst the editorial team this is my preferred option:

> JOSS may want to discourage using the official template (and JOSS logo) in preprints, and ask authors to convert to TeX on their own. IMO this is entirely reasonable, but if this is the case, it should be made clear.

I think it's reasonable to say that JOSS supports people uploading a preprint of their paper to the arXiv (and other preprint servers) but I think this has also caused confusion when the paper looks very similar to the published version in JOSS.

In summary my position is:

- Preprints are fine (in fact, encouraged!)
- JOSS discourages authors to use the official template for their preprints.

This second point seems to point towards it being a bad idea for us to support exporting (from @editorialbot) a TeX version of the paper.

@tarleb @xuanxu – perhaps we could have some way to produce a 'plain' version of a JOSS paper that is stripped of all of the JOSS styling (i.e., akin to the default Panda output). That should be possible right?

**Comment by tarleb** [comment]
I'll try to improve the support for that in Inara.

**Comment by xuanxu** [comment]
+1 to a `--plain` option

**Comment by szhorvat** [comment]
> I think it's reasonable to say that JOSS supports people uploading a preprint of their paper to the arXiv (and other preprint servers) but I think this has also caused confusion when the paper looks very similar to the published version in JOSS.

I fully agree with this. The first time I saw such a preprint as a reviewer, I though that it looked as if it has already been published. But then I checked the preprint policy, which clearly encourages preprints, so I did not mention anything about this in the review.

**Comment by tarleb** [comment]
I've opened https://github.com/openjournals/inara/pull/16. We may have to strip down the resulting LaTeX a little more. We'll also have to adjust the `openjournals/paperdraft` action and add more documentation.

**Comment by xuanxu** [comment]
📢 New command available for the JOSS reviews 📢

`@editorialbot generate preprint` will respond with a link to a simple _.tex_  version of the paper to be used in arXiv or other preprint servers.

The preprint file can also be created directly running the inara image with the following option: `-o preprint`

**Comment by arfon** [comment]
Amazing, thank you @xuanxu and @tarleb!

**Comment by kavanase** [comment]
This editorialbot preprint command is very useful! 

Just a note, it might be worth adding a mention of this to the `Preprint Policy` part of the JOSS submission guidelines (https://joss.readthedocs.io/en/latest/submitting.html#preprint-policy). Just it took a bit of Googling and looking for other options before I came across this thread. I realise now if I searched 'preprint' in the JOSS docs, I would've found this command (and I probably should have tried this), but it would make it a bit easier for authors if this was mentioned in the submission guidelines part. Just a suggestion!
